dlt/destinations/sql_client.py

```diff
@@ -274,6 +286,39 @@ def _truncate_table_sql(self, qualified_table_name: str) -> str:
         else:
             return f"DELETE FROM {qualified_table_name} WHERE 1=1;"
 
+    def iter_df(
+        self,
+        *,
+        table: str = None,
+        batch_size: int = 1000,
+        sql: str = None,
+        prepare_tables: List[str] = None,
+    ) -> Generator[DataFrame, None, None]:
+        if not sql:
+            table = self.make_qualified_table_name(table)
+            sql = f"SELECT * FROM {table}"
+
+        # iterate over results in batch size chunks
+        with self.execute_query(sql) as cursor:
+            while True:
+                if not (result := cursor.fetchmany(batch_size)):
```

Comment:
Ideally we immediately receive Arrow data from the SQL database. Columnar-to-row-to-columnar conversions will slow things down a lot.